packages/atomic/src/components.d.ts

```diff
@@ -356,6 +388,9 @@ declare global {
         "atomic-sort-dropdown": HTMLAtomicSortDropdownElement;
         "atomic-tab": HTMLAtomicTabElement;
         "atomic-text": HTMLAtomicTextElement;
+        "base-facet": HTMLBaseFacetElement;
```

Comment:
I'm wondering if there's a way to mark these as "internal", or that they should not be used on their own from an end user point of view ?

Or is that something that we might have to fix with some documentation shenanigans 🤔

Reply:
Another approach for this might have been to make them simple "classes" (more like... partial render function)/

It could be a nice experiment ? 

I'm not even sure if it would actually be better in the end, but... it seems more natural to me that we would have a bunch of utility render function that takes in some arguments as input (a headless facet search controller, for example), and that simply returns some JSX code.

What does everybody think ?

Reply:
Yes agreed IMO it shouldn't be fully feature stencil components (using the @Component decorator) we could perhaps look into functional components instead https://stenciljs.com/docs/functional-components#working-with-functional-components or utility classes like the combobox

Reply:
I agree it would be cleaner to hide Base components from end-users to start. Classes or functions both seem like good approaches.

Reply:
I looked into hiding components and didn't see an easy way to do so. I think using a functional component make sense for the `facet-value` class which is just basically rendering an HTML element for a given value. The `base-facet` component on the other hand manages all the state for the modal internally, so I don't think it can be functional without losing a lot of it's value.

Ideally we would want to be able to hide internal components and we should try and use functional components where possible, but I don't think that makes sense in all situations.